### PR TITLE
chore: move bash tests to integration

### PR DESCRIPTION
**Pull request by frezbo**
move extensions and secureboot tests to integration. Makes it easier to test.

### CODE REVIEW

File: hack/test/extensions/extension-patch-filter.jq [hunk 1]
```diff
@@ -4,7 +4,7 @@
         "path": "/machine/install/extensions",
         "value": [
             {
-                "image": map(select(. | contains("nvidia-container-toolkit") or contains("nvidia-fabricmanager") | not)) | .[]
+                "image": map(select(. | contains("nvidia") or contains("tailscale") | not)) | .[]
```
Comment (frezbo) [inline hack/test/extensions/extension-patch-filter.jq:7]: tailscale keeps on restarting without auth and nvida will be tested with another e2e

File: Makefile [hunk 2]
```diff
@@ -16,7 +16,7 @@ CLOUD_IMAGES_EXTRA_ARGS ?= ""
 
 ARTIFACTS := _out
 TOOLS ?= ghcr.io/siderolabs/tools:v1.5.0
-PKGS ?= v1.5.0
+PKGS ?= v1.5.0-1-g8a2227d
```
Comment (smira) [inline Makefile:19]: `main` should be using `1.6.0-alpha.0` tags...

File: internal/integration/api/common.go [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+	for _, expectedModule := range expectedVirtIOModules {
+		suite.Require().Contains(loadedModules, expectedModule, "expected module %s to be loaded", expectedModule)
+	}
+}
+
+// TestCommonDefaults verifies that the default ulimits are set.
+func (suite *CommonSuite) TestCommonDefaults() {
+	expectedUlimit := `
+core file size (blocks)         (-c) 0
```
Comment (smira) [inline internal/integration/api/common.go:98]: are these values stable with different RAM config?

File: internal/integration/api/reset.go [hunk 4]
```diff
@@ -218,9 +218,9 @@ func (suite *ResetSuite) TestResetWithSpecState() {
 		suite.T().Skip("cluster doesn't support reboot (and reset)")
 	}
 
-	if suite.Capabilities().TrustedBoot {
-		// this is because with trusted boot the talos.config is only applied and cannot be passed as kernel args
-		suite.T().Skip("cluster has trusted boot enabled, doesn't support reset")
+	if suite.TrustedBoot {
```
Comment (smira) [inline internal/integration/api/reset.go:221]: I think the previous made more sense - if we detect secure boot, we skip the tests, vs. that we were asked to verify that trusted boot is enabled.

File: internal/integration/api/trustedboot.go [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+func (suite *TrustedBootSuite) TearDownTest() {
+	if suite.ctxCancel != nil {
+		suite.ctxCancel()
+	}
+}
+
+// TestTrustedBootState verifies that the system is booted in secure boot mode
+// and that the disks are encrypted.
+func (suite *TrustedBootSuite) TestTrustedBootState() {
+	if testing.Short() {
```
Comment (smira) [inline internal/integration/api/trustedboot.go:54]: this should be a short test, so we could run it always?

File: internal/integration/api/trustedboot.go [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+	}
+
+	if !suite.TrustedBoot {
+		suite.T().Skip("skipping as talos is not explicitly trusted booted")
+	}
+
+	node := suite.RandomDiscoveredNodeInternalIP()
+	ctx := client.WithNode(suite.ctx, node)
+
+	securityResource, err := safe.StateGetByID[*runtimeres.SecurityState](ctx, suite.Client.COSI, runtimeres.SecurityStateID)
```
Comment (smira) [inline internal/integration/api/trustedboot.go:65]: in all these cases, better to use `safe.StateWatchFor` with types `Created, Updated`, as this doesn't depend on the resource being there at the moment of the test run

File: internal/integration/api/common.go (hunk 3 above)
Comment (frezbo) [reply internal/integration/api/common.go:98]: yes, works fine locally and in ci

File: internal/integration/api/trustedboot.go (hunk 5 above)
Comment (frezbo) [reply internal/integration/api/trustedboot.go:54]: hmm docker also uses short, so we'd then have to also check the provisioner used

File: internal/integration/api/reset.go (hunk 4 above)
Comment (frezbo) [reply internal/integration/api/reset.go:221]: didn't get that

Comment (smira) [reply internal/integration/api/reset.go:221]: in one of the previous iterations of this PR, you had `Capabilities().TrustedBoot` driven by the `SecurityState` resource, I remember reviewing it.

what I mean that this test should be skipped if the cluster is running in trusted boot mode, no matter if the flag `-test.trustedboot` was specified or not.

File: internal/integration/api/reset.go [hunk 7]
```diff
@@ -114,17 +114,9 @@ func (suite *ResetSuite) TestResetNodeByNode() {
...
-	}
-
-	if suite.Capabilities().TrustedBoot {
-		// this is because with trusted boot the talos.config is only applied and cannot be passed as kernel args
-		suite.T().Skip("cluster has trusted boot enabled, doesn't support reset")
-	}
-
-	if suite.Cluster == nil {
-		suite.T().Skip("without full cluster state reset test is not reliable (can't wait for cluster readiness in between resets)")
+	if !suite.Capabilities().SecureBooted {
```
Comment (smira) [inline internal/integration/api/reset.go:117]: the condition is inverted here and other places

Comment (frezbo) [reply internal/integration/api/reset.go:117]: ahh :face_with_head_bandage: , I'm dumb

**Pull request merged**